src/OpcUaServer/Server/DiscoveryClient.cpp

```diff
@@ -68,6 +68,21 @@ namespace OpcUaServer
 			return false;
 		}
 
+		// check if the DiscoveryUrl matches EndPointUrls
```

Comment:
I looked in the OPC UA specification. An OPC UA server may offer an OPC UA server and an OPC UA discovery server via the same port. Therefore, this test is not correct here. There must be another error here.

Reply:
Question:
Where does the OPC UA server crash?

Reply:
I think the actual problem is described in #322.

Reply:
This Issue will not fix the actuall error.  The error can be resolved with Issue #322. For this reason, this issue and its associated PR can be closed.

Reply:
Yes, the server crashes because the discovery client is not correct. So I described the error in #322. But I don't see any point in the configuration then the server registers itself to itself. What is why I made this PR. I agree that sever can play two roles as an application server and a discovery server at the sane time. But it must provide discovery services for other servers, not for itself.

Reply:
There is currently no interface between the OPC UA server and the Discovery Server application. Therefore, an error is built in with this PR. An OPC UA client can no longer find the OPC UA server.

Reply:
The OPC UA Discovery Server is provided in a separate application and in a separate git repo.

Reply:
I've reverted the commit with the check.